File: src/main/java/com/galaxy/restaurant/information/system/common/UserRole.java
package com.galaxy.restaurant.information.system.common;

import org.springframework.security.core.GrantedAuthority;

public enum UserRole implements GrantedAuthority {
    MANAGER, USER, ADMIN;

    @Override
    public String getAuthority() {
        return "ROLE_" + name();
    }
}

File: src/main/java/com/galaxy/restaurant/information/system/user/entity/UserRole.kt
package com.galaxy.restaurant.information.system.user.entity

import org.springframework.security.core.GrantedAuthority

enum class UserRole : GrantedAuthority {
    USER,
    MANAGER,
    ADMIN,
    ;

    override fun getAuthority(): String {
        return "ROLE_$name"
    }
}
